File: assignment-solutions/src/InterviewQues/Easy/PlusOne.java
package InterviewQues.Easy;

import java.util.Arrays;

public class PlusOne {
    public static void main(String[] args) {
        int[] arr = {9,9};
        System.out.println(Arrays.toString(plusOne(arr)));
    }

    static int[] plusOne(int[] arr) {
        for (int i = arr.length - 1; i >= 0; i--) {
            if (arr[i] < 9) {
                arr[i] = arr[i] + 1;
                return arr;
            } else {
                arr[i] = 0;
            }
        }
        int[] ans = new int[arr.length + 1];
        ans[0] = 1;
        return ans;
    }
}

File: assignment-solutions/src/Practice/Cpp/PlusOne.cpp
#include <iostream>
#include <vector>
using namespace std;

class PlusOne {
public:
    vector<int> plusOne(vector<int>& arr) {
        int n = arr.size();
        for (int i = n - 1; i >= 0; i--) {
        if (arr[i] < 9) {
            arr[i] = arr[i] + 1;
            return arr;
        } else {
            arr[i] = 0;
        }
    }
    
    vector<int> ans(n + 1, 0);
    ans[0] = 1;
    return ans;
    }
};

int main () {
    vector<int> arr = {1,1,9};
    PlusOne PlusOne;
    vector<int> ans = PlusOne.plusOne(arr);
    for (int i = 0; i < ans.size(); i++) {
        cout << ans[i] << " ";
    }
    return 0;
}